### ISSUE TITLE
More precise matching

### ISSUE THREAD

**Issue by nanozuki** [nanozuki/dantalian#16]
Search the tv anime "小魔女学园" in 2017 will match the movie "小魔女学园 魔法游行"; Search "リトルウィッチアカデミア" will match the OVA "小魔女学园" in 2013.

So, 
* We need a more precise matching method, maybe the year, even the month.
* Should add some method to specify anime manually.

**Comment by laxect** [comment]
So why not use tvshow.nfo(

**Comment by nanozuki** [comment]
@laxect Parse XML and let a user write XML is too complexy. A user-friendly and program-friendly config file is better.
I plan to define a `dantalian.toml` in every anime folder.

**Comment by nanozuki** [comment]
* We need a more precise matching method, maybe the year, even the month:

User can add year or month in folder name:

`小魔女学园 (2017)[BDRip]`
`小魔女学园 (201701)`

**Comment by nanozuki** [comment]
* Should add some method to specify anime manually：

dantalian.toml in every folder:

```toml
subject = 185792
url = "https://bangumi.tv/subject/185792"
tv_anime_pattern = "^(?P<name>.+?) (?P<date>\(\d{4,6}\))(?P<tags> (\[[^\s]+\])+)?$"
episode_pattern = '^AnimeName (?P<sp>SP)?(?P<ep>[_\d]+)\..*$"
```
* Subject id or URL to specify subject in bangumi.
* tv_anime_patter is a regular expression with named captures, "name" to specify anime name, "date" for the premiered date, `yyyy` or `yyyymm`, "tags" for custom tags.
* epidode_patter is a regular expression with named captures, "sp" to specify whether it's SP, "ep" for episode number.

**Comment by laxect** [comment]
> * We need a more precise matching method, maybe the year, even the month:
> 
> User can add year or month in folder name:
> 
> `小魔女学园 (2017)[BDRip]`
> `小魔女学园 (201701)`

Just use bgm id?

**Comment by nanozuki** [comment]
@laxect you mean subject id in folder name? `小魔女学园 (185792)`. Sounds good.

**Comment by nanozuki** [comment]
closed by #25
